File: src/2374. Node With Highest Edge Score.cpp
class Solution {
    public:
        int edgeScore(vector < int > & edges) {
            int n = edges.size();
            vector < long > scores(n, 0);
            for (int i = 0; i < n; ++i) {
                scores[edges[i]] += i;
            }
            int result;
            long maxScore = LONG_MIN;
            for (int i = 0; i < n; ++i) {
                if (scores[i] > maxScore) {
                    maxScore = scores[i];
                    result = i;
                }
            }
            return result;
        }
};

File: src/2374. Node With Highest Edge Score.js
/**
 * @param {number[]} edges
 * @return {number}
 */

export function edgeScore(edges) {
  const n = edges.length,
    scores = Array(n).fill(0)
  for (let i = 0; i < n; i++) {
    scores[edges[i]] += i
  }
  let result,
    max = -Infinity
  for (let i = 0; i < scores.length; i++) {
    if (scores[i] > max) {
      max = scores[i]
      result = i
    }
  }
  return result
}
